src/rust/engine/fs/src/lib.rs

```diff
@@ -764,12 +948,21 @@ pub trait VFS<E: Send + Sync + 'static>: Clone + Send + Sync + 'static {
     &self,
     path_glob: PathGlob,
     exclude: &Arc<Gitignore>,
-  ) -> BoxFuture<(Vec<PathStat>, Vec<PathGlob>), E> {
-    match path_glob {
+  ) -> BoxFuture<SingleExpansionResult, E> {
+    // TODO(cosmicexplorer): I would love to do something like `glob @ PathGlob::Wildcard {...}`,
```

Comment:
Thinking about this now, it makes sense, I think to disallow:
```rust
match path_glob {
  glob @ PathGlob::Wildcard { canonical_dir, symbolic_path, wildcard } => {
    path_glob
  }
}
```

Because you're moving `path_glob` into the destructured `canonical_dir`, `symbolic_path`, and `wildcard`, but then trying to move it again into the result of the expression -- there needs to be a clone happening somewhere. However if we were to match on `&path_glob` (which I just tried), it seems like it's impossible to move something while borrowing it, even if the move happens at the end of the borrowed scope. That seems like it doesn't need to happen? I'm not sure if my analysis is correct.

Reply:
Yeah; your analysis is correct that there isn't a nice way to make this work, and the reasons why. I'd quite like the language to somehow support (maybe as a special case), some syntax like:
```
match path_globs {
  PathGlob::Wildcard => path_globs
}
```

or

```
match path_globs {
  wildcard: PathGlob::Wildcard => wildcard
}
```

or something similar, to allow this, because it's a pretty common desire.